Microsoft.HealthVault/Clients/Deserializers/ThingDeserializer.cs

```diff
@@ -19,11 +21,24 @@ namespace Microsoft.HealthVault.Clients.Deserializers
 {
     internal class ThingDeserializer : IThingDeserializer
     {
+        private readonly Guid _applicationSpecificId = new Guid("a5033c9d-08cf-4204-9bd3-cb412ce39fc0");
+
         private readonly IHealthVaultConnection _connection;
 
-        public ThingDeserializer(IHealthVaultConnection connection)
+        private readonly IThingTypeRegistrar _thingTypeRegistrar;
+
+        private readonly Dictionary<string, Dictionary<string, Type>> _appSpecificHandlers;
+        private readonly Dictionary<Guid, Type> _typeHandlers;
+        private readonly Dictionary<string, Type> _extensionHandlers;
+
+        public ThingDeserializer(IHealthVaultConnection connection, IThingTypeRegistrar thingTypeRegistrar)
         {
             _connection = connection;
+            _thingTypeRegistrar = thingTypeRegistrar;
+
+            _appSpecificHandlers = _thingTypeRegistrar.RegisteredAppSpecificHandlers;
+            _typeHandlers = _thingTypeRegistrar.RegisteredTypeHandlers;
+            _extensionHandlers = _thingTypeRegistrar.RegisteredExtensionHandlers;
```

Comment:
good one ..  I favor reading always from thingtyperegistrar .. 

---
In reply to: [123320069](https://github.com/Microsoft/healthvault-dotnetstandard-sdk/pull/62#discussion_r123320069) [](ancestors = 123320069)